test/generate_ssl_tests.pl

```diff
@@ -43,11 +43,24 @@ sub print_templates {
     # Add the implicit base configuration.
     foreach my $test (@ssltests::tests) {
         $test->{"server"} = { (%ssltests::base_server, %{$test->{"server"}}) };
-	# Do not emit an empty "server2" section.
-	if (defined $test->{"server2"}) {
+        if (defined $test->{"server2"}) {
             $test->{"server2"} = { (%ssltests::base_server, %{$test->{"server2"}}) };
+        } elsif (defined $test->{"test"}->{"ServerNameCallback"}) {
+            # Default is the same as server.
+            $test->{"server2"} = { (%ssltests::base_server, %{$test->{"server"}}) };
         } else {
-            $test->{"server2"} = { };
+            # Do not emit an empty "server2" section.
+            $test->{"server2"} = { }
```

Comment:
Yes.  In perl, the semi is a statement *separator*, not a statement *terminator* (like it is in C).